aten/src/ATen/ParallelNativeMobile.h

```diff
@@ -0,0 +1,121 @@
+#pragma once
+
+#include <algorithm>
+#include <cstddef>
+#include <exception>
+
+#define INTRA_OP_PARALLEL
+
+namespace at {
+namespace internal {
+
+// Run lambda function `fn` over `task_id` in [0, `range`) with threadpool.
+// `fn` will be called with params: (thread_pool_task_id, task_id).
+CAFFE2_API void _run_with_pool(const std::function<void(int, size_t)>& fn, size_t range);
+
+// RAII guard helps to support in_parallel_region() and get_thread_num() API.
+struct CAFFE2_API ParallelRegionGuard {
+  ParallelRegionGuard(int64_t task_id);
+  ~ParallelRegionGuard();
+};
+
+inline std::tuple<size_t, size_t> calc_num_tasks_and_chunk_size(
```

Comment:
@ilia-cher let me know if I should move this to a more common place and share with ParallelNative.h.

Reply:
ParallelCommon could be this place, although doesn't seem to be needed in the merged version